Source/WebCore/rendering/RenderCounter.cpp

```diff
@@ -466,26 +466,27 @@ String RenderCounter::originalText() const
     RefPtr child = m_counterNode.get();
     int value = child->actsAsReset() ? child->value() : child->countInParent();
 
-    auto counterText = [](const ListStyleType& styleType, int value, CSSCounterStyle* counterStyle) {
+    auto counterText = [](const ListStyleType& styleType, int value, CSSCounterStyle* counterStyle, TextDirection textDirection) {
         if (styleType.type == ListStyleType::Type::None)
             return emptyString();
 
         if (styleType.type == ListStyleType::Type::CounterStyle) {
             ASSERT(counterStyle);
-            return counterStyle->text(value);
+            return counterStyle->text(value, textDirection);
         }
 
         ASSERT_NOT_REACHED();
         return emptyString();
     };
     auto counterStyle = this->counterStyle();
-    String text = counterText(m_counter.listStyleType(), value, counterStyle.get());
+    auto textDirection = style().isLeftToRightDirection() ? TextDirection::LTR : TextDirection::RTL;
```

Comment:
style().direction() should give you directly what you need instead of the ternary